tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -332,9 +332,71 @@ private void AsmDataProductConfigChanged(object sender, ProductConfigChangedEven
             }
         }
 
+        private void AsmProductConfigChanged(object sender, ProductConfigChangedEventArgs e)
+        {
+            if (!_enabled) { return; }
+
+            var asmConfigs = e.GetDeserializedConfigurations<RcmModels.Asm.Payload>();
+            int ruleCount = 0;
+            foreach (var asmConfig in asmConfigs)
+            {
+                try
+                {
+                    var rulesStatus = asmConfig.TypedFile.RuleStatus;
+                    lock (_ruleStatus)
+                    {
+                        foreach (var data in rulesStatus)
+                        {
+                            if (data.Id == null || data.Enabled == null)
+                            {
+                                continue;
+                            }
+
+                            _ruleStatus[data.Id] = data.Enabled.Value;
```

Comment:
Is the `RuleStatus`/`rule_overrides` a complete representation? Or is it a delta from the previous message? If it's a delta, then this approach looks good. If instead it's a complete representation, then I would think we'd want to create a brand new dictionary each time and replace `_ruleStatus` with the new dictionary

Reply:
I assumed we receive deltas on each message. But this point is not clear in the docs